pkg/pbq/pbq.go

```diff
@@ -0,0 +1,157 @@
+package pbq
+
+import (
+	"context"
+	"errors"
+	"github.com/numaproj/numaflow/pkg/isb"
+	"github.com/numaproj/numaflow/pkg/pbq/store"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"go.uber.org/zap"
+	"time"
+)
+
+var COBErr error = errors.New("error while writing to pbq, pbq is closed")
+var EOF error = errors.New("error while reading, EOF")
+
+type PBQ struct {
+	Store       store.Store
+	output      chan *isb.Message
+	cob         bool // cob to avoid panic in case writes happen after close of book
+	partitionID string
+	options     *store.Options
+	isReplaying bool
+	manager     *Manager
+	log         *zap.SugaredLogger
+}
+
+// NewPBQ accepts size and store and returns new PBQ
+func NewPBQ(ctx context.Context, partitionID string, persistentStore store.Store, pbqManager *Manager, options *store.Options) (*PBQ, error) {
```

Comment:
here we do not require `store.Options` since we are passing in `store.Store`, it requires PBQ options.

Reply:
done